playground/src/components/state-machines-list/state-machines-list.js

```diff
@@ -35,107 +34,84 @@ export const columnsDefinition = [
 const getErrorMessage = error =>
   error.stack || error.message || error.toString();
 
-export class StateMachinesList extends React.Component {
-  static displayName = 'StateMachinesList';
-  static propTypes = {
-    goToStateMachineDetail: PropTypes.func.isRequired,
-    projectKey: PropTypes.string.isRequired,
-    // injected
-    applicationContext: PropTypes.shape({
-      dataLocale: PropTypes.string.isRequired,
-    }).isRequired,
-    cachedStateMachineObjectsCount: PropTypes.number,
-  };
-  measurementCache = null;
-  registerMeasurementCache = cache => {
-    this.measurementCache = cache;
-  };
-  handleRowClick = (rowIndex, results) => {
-    this.props.goToStateMachineDetail(results[rowIndex].id);
-  };
+const StateMachinesList = props => {
+  const cachedStateMachine = useSelector(selectStateMachinesFromCache);
+  const cachedStateMachineObjectsCount = cachedStateMachine
```

Comment:
Isn't this just `hasStateMachines` as a better name?

Reply:
Hmm what do you mean? The `cachedStateMachineObjectsCount` is a number.